pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -73,11 +73,16 @@ def restore_weights(self, model: LightningModule):
         if self.trainer.on_gpu:
             torch.cuda.empty_cache()
 
-    def restore(self, checkpoint_path: str, on_gpu: bool):
+    def restore(self, checkpoint_path: str, on_gpu: bool) -> bool:
         """
         Load model/training states from a 'PyTorch-Lightning checkpoint' file through file-read and state-restore.
         All restored states are listed in return value description of `dump_checkpoint`.
         """
+        # Try to read the checkpoint file at `checkpoint_path`. If not exist, do not restore checkpoint.
+        fs = get_filesystem(checkpoint_path)
+        if not fs.exists(checkpoint_path):
+            rank_zero_warn("No checkpoint file exists at `resume_from_checkpoint`. Start from scratch")
+            return False
```

Comment:
better check this during `Trainer.init` itself??

Reply:
It can be one option, but I have no confidence about immutability of `checkpoint_path`.  
If it is mutable, `Trainer.init` is not suitable.  
In my opinion, separated refactoring PR is best for this problem.

Reply:
sorry I didn't get what you mean by `immutability of checkpoint_path`. can you give an example or something?

Reply:
ok got it :+1: read the issue just now.